bundle/edu.gemini.itc/src/main/java/edu/gemini/itc/shared/EmissionLineSpectrum.java

```diff
@@ -1,71 +1,53 @@
 package edu.gemini.itc.shared;
 
+import edu.gemini.spModel.core.Wavelength;
+import edu.gemini.spModel.target.EmissionLine;
+
 /**
  * This class creates a EmissionLine spectrum over the interval defined by the
  * blocking filter.  The code comes from PPuxley's Mathcad Demo
  * This Class implements Visitable sampled specturm to create the sed.
  */
 
-public class EmissionLineSpectrum implements VisitableSampledSpectrum {
-    private DefaultSampledSpectrum _spectrum;
+public final class EmissionLineSpectrum implements VisitableSampledSpectrum {
+
+    private final DefaultSampledSpectrum _spectrum;
 
     // Private c'tor to support clone()
-    private EmissionLineSpectrum(DefaultSampledSpectrum spectrum) {
+    private EmissionLineSpectrum(final DefaultSampledSpectrum spectrum) {
         _spectrum = spectrum;
     }
 
-    public EmissionLineSpectrum(double wavelength, double width, double flux,
-                                double continuumFlux, String lineFluxUnits,
-                                String continuumFluxUnits, double z, double interval) {
-        double _flux;
-        // convert the wavelenght to Nanometers.
-        double _wavelength = wavelength * 1000;
-        double _sampling = interval;  //0.2
-        double lambda;  // var to be used to construct sed
-
-        double start = 300;//0.2*_wavelength;//.8
-        double end = 30000;//1.8*_wavelength;//1.2
-        //System.out.println("Start: "+ start +"End: " +end);
+    public EmissionLineSpectrum(final Wavelength wavelength, final double width, final EmissionLine.Flux flux,
+                                final EmissionLine.Continuum continuum, final double z, final double interval) {
 
         //shift start and end depending on redshift
-        start /= (1 + z);
-        end /= (1 + z);
+        final double start = 300 / (1 + z);
```

Comment:
Maybe you could preserve (adapt) the comments from the deleted lines as the formula is not that obvious